### ISSUE TITLE
Importing LXML fails (Python 3.6)

### ISSUE THREAD

**Issue by ncurwen** [nficano/python-lambda#60]
Hi everyone,

I'm working on a lambda that will makes use of LXML. When running the lambda in AWS I noticed the following error message: `Unable to import module 'service': cannot import name 'etree'`. It seems this is because "LXML must be built with C extensions for libxml2 and libxslt in a way that plays well with the Amazon Lambda execution environment."

Has anyone else run into this issue?

source: https://www.azavea.com/blog/2016/06/27/using-python-lxml-amazon-lambda/

**Comment by jmeekhof** [comment]
Yes, I've run into this type of issue before. You have 2 issues. 

1. Your c extension must be compiled for Amazon linux. There are 2 ways to solve this. 
a. spin up an ec2 instance and deploy from there.
b. create a docker container and develop and deploy from the container.(this is how I do it) Here's a quick and dirty Dockerfile: https://gist.github.com/jmeekhof/f226a8e148f345c71e8c0f87123ee8bf

2. Your second issue is you need to be able to deploy your compiled object with your lambda function. Right now, this project doesn't have a way to automate the additional objects.  Your best workaround is to `lambda build` and then using a zip utility add the necessary shared libraries, and manually deploy. Here's a good write up on this process: http://www.perrygeo.com/running-python-with-compiled-code-on-aws-lambda.html

**Comment by ncurwen** [comment]
Hi Josh, I appreciate the help.

I'm working on the Docker solution you mentioned but I'm running into some issues with permissions within the container. When execution `lambda build` within the container it's unable to create the .zip due to not having write permissions in the mounted `code` directory.

Any suggestions?

Thanks,
Nick

**Comment by jmeekhof** [comment]
Nick,

So the volume you're mounting into your code directory should be a real directory on your host machine. If it's not real or you don't have permissions for it on your host, it won't work. Can you share the docker command you are using to run this?

```bash

**Comment by ncurwen** [comment]
Josh,

I've got permissions for the local code directory and it's the proper path to the root of the project (doing an `ls` within the container I can see the contents). Here's the command.

```
docker run -it -v \
~/Projects/fdd-collector:/home/dev/code \
-v ~/.aws:/home/dev/.aws \
-v ~/.gitconfig:/home/dev/.gitconfig \
simple.lambda
```

I was doing some more research and it seems that it's probably due to my Docker setup. My team is using an older flavour of Docker (Boot2Docker) for some other projects which is likely where the issue is coming from. We really need to update to Docker for Mac but there were some complications and performance issues the last time we tried so we've been putting it off.

Thanks again,
Nick
